File: src/main/java/org/team1/nbe1_3_team01/domain/board/entity/Category.java
package org.team1.nbe1_3_team01.domain.board.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.team1.nbe1_3_team01.domain.group.entity.Team;

@Entity
@Getter
@Table(name = "board_category")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 50)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;

    @Builder
    private Category(Team team,
                     String name) {
        this.team = team;
        this.name = name;
        team.addCategory(this);
    }

}

File: src/main/java/org/team1/nbe1_3_team01/domain/board/entity/Category.kt
package org.team1.nbe1_3_team01.domain.board.entity

import jakarta.persistence.*
import org.team1.nbe1_3_team01.domain.group.entity.Team

@Entity
@Table(name = "board_category")
data class Category (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long = 0L,

    @JoinColumn(name = "team_id")
    @ManyToOne(fetch = FetchType.LAZY)
    private val team: Team,

    @Column(length = 50)
    private var name: String
) {

    init {
        team.addCategory(this)
    }
}
